docs/config/worker-options.md

```diff
@@ -11,9 +11,10 @@ Output format for worker bundle.
 
 ## worker.plugins
 
-- **Type:** [`(Plugin | Plugin[])[]`](./shared-options#plugins)
+- **Type:** [`() => (Plugin | Plugin[])[]`](./shared-options#plugins)
 
-Vite plugins that apply to worker bundle. Note that [config.plugins](./shared-options#plugins) only applies to workers in dev, it should be configured here instead for build.
+Vite plugins that apply to the worker bundles. Note that [config.plugins](./shared-options#plugins) only applies to workers in dev, it should be configured here instead for build.
+The function should return new plugin instances as they are used in parallel rollup worker builds. As such, modifying `config.worker` options in the `config` hook will be ignored.
```

Comment:
So... for users that wants to modify `config.worker` option should put that plugin in the top-level `plugins`, am I right?

Reply:
Yes, that is the idea. We could actually support this for some config fields but not for others so probably it is better to draw a hard line. I don't think the API is less flexible due to this, it makes sense modifications to the main config file should be done in the main plugins.